### ISSUE TITLE
Saying all videos are downloaded when they're not

### ISSUE THREAD

**Issue by kave333** [DIGITALCRIMINAL/OnlyFans#44]
When i run the script and let it run, it took around 1h 30mins to get 350~ videos from an onlyfans account then it finished and said it had grabbed them all. There is roughly 2600 videos on said only fans account. Shall i just re run the script with "overwrite_files" set to false and see if it grabs more videos and keep doing it till i all have been downloaded?

**Comment by DIGITALCRIMINAL** [comment]
Jesus, that's a lot. Try with overwriting on.
Also, some videos get downloaded into the image folder. OnlyFans changed them how their API works so I need to fix that. 😢

**Comment by kave333** [comment]
I've tried with overwriting on and off. I opened the .JSON file and it just shows 1 big line of everything so i filtered it to just find how many "https" there was in the whole .json file. It returned with 2,598 matches so im guessing it got the links to 2598 videos where on the OF pages it shows 2,680 so from this i guess some are private... 

I ran using the arguments B and C to try eliminate videos and images getting mixed. 

Images it managed to download 2404. on the OF page there is 15,981. Then i used the same method in the .JSON file and it shows there was 15,981 "https" matches so script is scraping the links correctly as shown in the JSON files but not all the links are getting downloaded.

**Comment by dilemmax** [comment]
What you can do is using the below link to convert the JSON file to an excel spreadsheet. You should be able to accurately find out how links are available for download.

you can copy the links to a download manager to make it download faster and it's also resumable as well. I'm using Internet Download Manager and it works really well.

https://json-csv.com/

**Comment by dilemmax** [comment]
Can you also do me a favour and sort the links by the Post ID and check the date column to see what is the earliest date for the links? I'm willing to be it stops before early 2017.

**Comment by DIGITALCRIMINAL** [comment]
@kave333 
I've fixed the image and video mixing issue but I'm not ready to push the commit yet.
I'm also overhauling the json exporting.

@dilemmax I found corrupted posts in 2017 that wouldn't download. It was due to them having the same filename. It WAS the same photo though but OnlyFans decided to duplicate them...

Since the script is multithreaded, the script downloads both files with the same name at the same time which causes them to overwrite each other and become corrupted. That's why you'll see fewer files than you accounted for. To fix this I'll just tell the script to search for duplicates and rename them and it should solve the problem.

**Comment by DIGITALCRIMINAL** [comment]
I also found a problem with the math in my script that would get 1 page less when rounding to the floor.
Fixed this as well. Few more tests and I'll push a commit.

**Comment by DIGITALCRIMINAL** [comment]
Pushed the commit. Open/reopen if you still got a problem.

**Comment by kave333** [comment]
@dilemmax anyway i can contact you to discuss the IDM further?

**Comment by dilemmax** [comment]
@kave333 sure can. my email address is dilemmax@gmail.com

**Comment by dilemmax** [comment]
> I also found a problem with the math in my script that would get 1 page less when rounding to the floor.
> Fixed this as well. Few more tests and I'll push a commit.

Thanks DigitalCriminal, I'll test it out now

**Comment by dilemmax** [comment]
@DigitalCriminal, I don't think I can re-open this issue since I'm not the OP. I'm still having the same problem. It's still scrapping the same video links but I'm think maybe the Content Creator may posted the videos at some stage and then made it private? Just a theory.

**Comment by DIGITALCRIMINAL** [comment]
Hmm, what's the total number of videos on the account @dilemmax

**Comment by dilemmax** [comment]
312 videos but it only scrapped 307 video links

**Comment by DIGITALCRIMINAL** [comment]
@dilemmax have you checked the archive.json to check if there are any invalid links?

**Comment by dilemmax** [comment]
Is archive.json in the same folder as links.json?

**Comment by dilemmax** [comment]
okay thanks. Nope it didn't create one.

**Comment by DIGITALCRIMINAL** [comment]
Are you using the latest release because the script doesn't create links.json anymore

**Comment by dilemmax** [comment]
yeah I am. Sorry I meant videos.json

**Comment by DIGITALCRIMINAL** [comment]
@dilemmax also, my bad. The archive.json is for a different module and images/videos/audio.json is correct. It seems like we have the same problem. I can't my missing videos either. I'm going to look into it further now though to see what's happening.

**Comment by dilemmax** [comment]
Okay no worries. I wish I have enough programming knowledge to help you troubleshoot this problem.

**Comment by dilemmax** [comment]
Do you think it has something to do with pinned posts? There's 4 pinned posts in mine but 307+4 doesn't equal 312

**Comment by DIGITALCRIMINAL** [comment]
@dilemmax It's all good, you've helped alot c:
And nah it's not the pinned posts because mine is in the .json
I decided to count all the media using the posts api and I found 1938 out of 1939, so I am getting closer. (1938 is made out of  images and videos count)

**Comment by DIGITALCRIMINAL** [comment]
I honestly just think that the posts are private. The missing content isn't even in the API

**Comment by DIGITALCRIMINAL** [comment]
@dilemmax 
Added an option to export to csv instead of json in the latest release.
![image](https://user-images.githubusercontent.com/21285176/67220746-40847a00-f422-11e9-8e94-a849071765d0.png)

**Comment by dilemmax** [comment]
Thanks @DigitalCriminal! That would help a lot! While you're at it, is there anyway to strip away text in angle brackets and unicode text from the "text" column when scrapping to a json/csv file?

Currently, I open the json file in Word and use the wildcard search function to delete any text starting and ending with angle brackets, replace \n with a space etc. Then convert it to a spreadsheet and make the relevant changes before creating a batch file to rename the files with relevant titles.

Basically I have to do a lot before I end up with something like this 

"ren "5d9ec0518452aaf05bd2d.mp4" "[2019-10-11] 194 - HOOK UP P2 JAYJAY & MM ENJOY!.mp4""

**Comment by dilemmax** [comment]
Hope I make sense!

**Comment by DIGITALCRIMINAL** [comment]
@dilemmax open new issue for this please

**Comment by dilemmax** [comment]
Will do :)
